doc: minimal documentation of supported platforms (#254741)
NixOS/nixpkgs: doc/platform-support.appendix.md

# Platform Support {#appendix-platform-support}

> Currently, the `x86_64-linux` platform receives the highest level of support, both in terms of CI and of developer attention.
> The `aarch64-linux` plaform is also well supported, with most packages building succesfully in CI.
> 
> On Darwin, `aarch64-darwin` receives better support than `x86_64-darwin`.

Reviewer: ```suggestion
- `x86_64-linux`: Highest level of support, both in terms of CI and of developer attention.
- `aarch64-linux`: Well supported, with most packages building successfully in CI.
- `aarch64-darwin`: Receives better support over `x86_64-darwin`
- `x86_64-darwin`: Receives some support
```

Author: Do you find this better? I personally prefer the more discursive style, in this case.

Reviewer 2: Sometimes you do just want to scan through.

Could have the more verbose description and a more concise table showing support?

Reviewer 3: Also the end state is probably a full table (eventually), so we can start in this direction anyway?

Author: Sounds good, I'll switch to a table.

> On Darwin, `aarch64-darwin` receives better support than `x86_64-darwin`.

Reviewer 3: Isn't `x86_64-darwin` and `aarch64-darwin` situation more complicated than really comparable levels because from time to time CI before merging actually works better for the former?

Reviewer 4: @toonn may have a thought on this.

Reviewer 5: I can't really disagree with the statement. Most of NixOS/darwin-maintainers seem to be aarch64-darwin users. The Darwin builders for Hydra are aarch64-darwin for the most part, emulating x86_64-darwin when necessary. Maybe a qualifier like "slightly" is appropriate. The goal is for them to be equal for the foreseeable future.

> For a more formal definition of the different platform support tiers, see [RFC046]. Note though that the platform list in [Appendix A] is out of date.

Reviewer: ```suggestion
A more formal definition of different platform support tiers was accepted with [RFC046], though it has not been fully implemented so far.
```

Author: it's not clear what hasn't been implemented. I don't want readers to have to figure that out by reading the whole RFC. That's why I mentioned specifically the platform list.

Reviewer 4: ```suggestion
A more formal definition of different platform support tiers was accepted with [RFC046], but is not fully implemented so far.
The preliminary platform list in the RFC's [Appendix A] can be used as guidance, but is not up to date.
```

Reviewer: That sounds good to me
